reV/generation/generation.py

```diff
@@ -482,11 +489,85 @@ def time_index(self):
             elif step is not None:
                 time_index = time_index[::step]
 
-            self._time_index = self.handle_leap_ti(time_index,
-                                                   drop_leap=self._drop_leap)
+            time_index = self.handle_lifetime_index(time_index)
+            time_index = self.handle_leap_ti(time_index,
+                                             drop_leap=self._drop_leap)
+
+            self._time_index = time_index
 
         return self._time_index
 
+    def handle_lifetime_index(self, ti):
+        """Adjust the time index if modeling full system lifetime.
+
+        Parameters
+        ----------
+        ti : pandas.DatetimeIndex
+            Time-series datetime index with leap days.
+
+        Returns
+        -------
+        ti : pandas.DatetimeIndex
+            Time-series datetime index.
+        """
+        life_var = "system_use_lifetime_output"
```

Comment:
Can you please document this key and it's subtleties (i.e. every SAM config has to have this option enabled, and every SAM config has to have the same `analysis_period`, etc.) after the `clearsky` documentation here: https://github.com/NREL/reV/blob/0bf8745d5725e83d49a6a67570996a11d4a0f1aa/reV/SAM/generation.py#L817-L820

Reply:
Ah, I see this is a SAM key, not just a reV-specific key. In that case, can you create a section in the style that we have for reV Geothermal ("notable SAM keys"), and put the documentation there instead? https://github.com/NREL/reV/blob/0bf8745d5725e83d49a6a67570996a11d4a0f1aa/reV/SAM/generation.py#L1324-L1331

Reply:
Okay, wrote a description of the lifetime specific keys needed, check it out

Reply:
Looks great, thanks so much!!